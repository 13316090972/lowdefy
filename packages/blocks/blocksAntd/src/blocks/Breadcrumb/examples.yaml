- id: default
  type: Breadcrumb
- id: properties.list.label strings
  type: Breadcrumb
  properties:
    list:
      - Page One
      - Page two
- id: properties.list.pageId
  type: Breadcrumb
  properties:
    list:
      - pageId: page1
      - pageId: page2
- id: properties.list.icon
  type: Breadcrumb
  properties:
    list:
      - label: label 1
        icon: SaveOutlined
      - label: label 2
        icon: DeleteOutlined
- id: properties.list.icon with onClick
  type: Breadcrumb
  properties:
    list:
      - label: label 1
        icon: SaveOutlined
      - label: label 2
        icon: DeleteOutlined
  events:
    onClick: []
- id: properties.list.url
  type: Breadcrumb
  properties:
    list:
      - url: https://www.url.com
      - url: https://docs.url.com
- id: 'properties.list.pageId label and url'
  type: Breadcrumb
  properties:
    list:
      - url: https://www.url.com
        pageId: page1
      - url: https://www.url.com
        label: label url
      - pageId: page2
        label: label pageId
      - url: https://docs.url.com
        pageId: page3
      - label: label4
- id: properties.list.style
  type: Breadcrumb
  properties:
    list:
      - label: Page One
        style:
          color: blue
      - label: Page two
        style:
          color: red
- id: properties.separator
  type: Breadcrumb
  properties:
    separator: '->'
    list:
      - label: Page One
      - label: Page two
- id: 'properties.style: fontSize: 18'
  type: Breadcrumb
  properties:
    style:
      fontSize: 18
    list:
      - label: Page One
      - label: Page two
