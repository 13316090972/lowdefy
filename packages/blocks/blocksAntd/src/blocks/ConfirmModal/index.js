export default {
  import: {
    block: 'blocks/ConfirmModal/ConfirmModal.js',
    styles: ['blocks/ConfirmModal/style.less'],
  },
  meta: {
    category: 'container',
    loading: false,
  },
  test: {
    methods: [
      {
        name: 'open',
        args: {},
      },
      {
        name: 'open',
        args: {
          status: 'warning',
        },
      },
      {
        name: 'open',
        args: {
          message: 'Args message',
        },
      },
      {
        name: 'open',
        args: {
          description: 'Args description',
        },
      },
      {
        name: 'open',
        args: {
          duration: 1,
        },
      },
    ],
  },
};
