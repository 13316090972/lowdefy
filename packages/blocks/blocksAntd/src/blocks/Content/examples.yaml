- id: default
  type: Content
- id: properties.style
  type: Content
  properties:
    style: 
      border: '1px solid blue'
- id: areas.content
  type: Content
  areas:
    content:
      blocks:
        - id: testArea
          type: Test
- id: areas.footer
  type: Content
  areas:
    footer:
      blocks:
        - id: testArea
          type: Test
- id: areas.header
  type: Content
  areas:
    header:
      blocks:
        - id: testArea
          type: Test