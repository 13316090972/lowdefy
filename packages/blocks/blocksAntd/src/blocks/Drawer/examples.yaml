- id: default
  type: Drawer
- id: 'properties.title'
  type: Drawer
  properties:
    title: Drawer title
- id: 'properties.closable: false'
  type: Drawer
  properties:
    closable: false
- id: 'properties.mask'
  type: Drawer
  properties:
    mask: false
- id: 'properties.maskClosable: false'
  type: Drawer
  properties:
    maskClosable: false
- id: 'properties.width: 500'
  type: Drawer
  properties:
    width: 500
- id: 'properties.placement: top'
  type: Drawer
  properties:
    placement: top
- id: 'properties.height: 500 placement: top'
  type: Drawer
  properties:
    placement: top
    height: 500
- id: 'properties.zIndex: 1100'
  type: Drawer
  properties:
    zIndex: 1100
- id: 'properties.placement: bottom'
  type: Drawer
  properties:
    placement: bottom
- id: 'properties.placement: left'
  type: Drawer
  properties:
    placement: left
- id: 'properties.keyboard: false'
  type: Drawer
  properties:
    keyboard: false
- id: 'properties.contentWrapperStyle: border: 10px solid green'
  type: Drawer
  properties:
    contentWrapperStyle:
      border: 10px solid green
- id: 'areas.extra:'
  type: Drawer
  areas:
    extra:
      blocks:
        - id: one
          type: Test
