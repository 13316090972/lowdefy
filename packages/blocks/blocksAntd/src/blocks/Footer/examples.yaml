- id: default
  type: Footer
- id: properties.style
  type: Footer
  properties:
    style: 
      border: '1px solid blue'
- id: areas.content
  type: Footer
  areas:
    content:
      blocks:
        - id: testArea
          type: Test