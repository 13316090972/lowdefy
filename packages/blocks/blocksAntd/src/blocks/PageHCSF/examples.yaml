- id: default
  type: PageHCSF
- id: areas.content
  type: PageHCSF
  areas:
    content:
      blocks: []
- id: areas.footer
  type: PageHCSF
  areas:
    footer:
      blocks: []
- id: areas.header
  type: PageHCSF
  areas:
    header:
      blocks: []
- id: areas.sider
  type: PageHCSF
  areas:
    sider:
      blocks: []
- id: "properties.header.theme: light - dark is default"
  type: PageHCSF
  properties:
    header:
      theme: light
  areas:
    content:
      blocks: []
    footer:
      blocks: []
    header:
      blocks: []
    sider:
      blocks: []
- id: "properties.header.color: #4432a8 "
  type: PageHCSF
  properties:
    header:
      color: "#4432a8"
  areas:
    content:
      blocks: []
    footer:
      blocks: []
    header:
      blocks: []
    sider:
      blocks: []
- id: "properties.style: border: 5px solid blue"
  type: PageHCSF
  properties:
    style:
      border: 5px solid blue
  areas:
    content:
      blocks: []
    footer:
      blocks: []
    header:
      blocks: []
    sider:
      blocks: []
- id: "properties.content.style: border: 5px solid blue"
  type: PageHCSF
  properties:
    content:
      style:
        border: 5px solid blue
  areas:
    content:
      blocks: []
    footer:
      blocks: []
    header:
      blocks: []
    sider:
      blocks: []
- id: "properties.header.style: border: 5px solid blue"
  type: PageHCSF
  properties:
    header:
      style:
        border: 5px solid blue
  areas:
    content:
      blocks: []
    footer:
      blocks: []
    header:
      blocks: []
    sider:
      blocks: []
- id: "properties.footer.style: border: 5px solid blue"
  type: PageHCSF
  properties:
    footer:
      style:
        border: 5px solid blue
  areas:
    content:
      blocks: []
    footer:
      blocks: []
    header:
      blocks: []
    sider:
      blocks: []
- id: "properties.sider.style: border: 5px solid blue"
  type: PageHCSF
  properties:
    sider:
      style:
        border: 5px solid blue
  areas:
    content:
      blocks: []
    footer:
      blocks: []
    header:
      blocks: []
    sider:
      blocks: []
- id: "properties.main.style: border: 5px solid blue"
  type: PageHCSF
  properties:
    main:
      style:
        border: 5px solid blue
  areas:
    content:
      blocks: []
    footer:
      blocks: []
    header:
      blocks: []
    sider:
      blocks: []
