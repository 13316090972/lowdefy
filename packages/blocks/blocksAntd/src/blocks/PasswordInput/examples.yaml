- id: default
  type: PasswordInput
- id: 'properties.title: My PasswordInput Title'
  type: PasswordInput
  properties:
    title: My PasswordInput Title
- id: 'properties.autoFocus: true'
  type: PasswordInput
  properties:
    autoFocus: true
- id: 'properties.disabled: true'
  type: PasswordInput
  properties:
    disabled: true
- id: 'properties.placeholder: Type your text here'
  type: PasswordInput
  properties:
    placeholder: Type your text here
- id: 'properties.size: large'
  type: PasswordInput
  properties:
    size: large
- id: 'properties.size: small'
  type: PasswordInput
  properties:
    size: small
- id: 'properties.visibilityToggle: false'
  type: PasswordInput
  properties:
    visibilityToggle: false
- id: 'properties.bordered: false'
  type: PasswordInput
  properties:
    bordered: false
- id: 'properties.label span: 12'
  type: PasswordInput
  properties:
    label:
      span: 12
- id: 'properties.label align: right'
  type: PasswordInput
  properties:
    label:
      align: right
- id: 'properties.label inline: true'
  type: PasswordInput
  properties:
    label:
      inline: true
- id: 'properties.label disabled: true'
  type: PasswordInput
  properties:
    label:
      disabled: true
- id: 'properties.label colon: false'
  type: PasswordInput
  properties:
    label:
      colon: false
- id: 'properties.label extra: showing extra'
  type: PasswordInput
  properties:
    label:
      extra: showing extra
