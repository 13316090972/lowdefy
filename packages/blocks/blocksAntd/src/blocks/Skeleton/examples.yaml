# Copyright 2020-2021 Lowdefy, Inc

# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at

#     http://www.apache.org/licenses/LICENSE-2.0

# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

- id: default
  type: Skeleton
- id: properties.active
  type: Skeleton
  properties:
    active: true
- id: 'properties.title boolean'
  type: Skeleton
  properties:
    title: false
- id: 'properties.title object'
  type: Skeleton
  properties:
    title:
      width: 250
- id: 'properties.avatar boolean'
  type: Skeleton
  properties:
    avatar: true
- id: 'properties.avatar object circle'
  type: Skeleton
  properties:
    avatar:
      shape: circle
      size: large
- id: 'properties.avatar object square'
  type: Skeleton
  properties:
    avatar:
      shape: square
      size: small
- id: 'properties.button object circle'
  type: Skeleton
  properties:
    button:
      shape: circle
      size: large
- id: 'properties.button object round'
  type: Skeleton
  properties:
    button:
      shape: round
      size: small
- id: 'properties.block block true'
  type: Skeleton
  properties:
    button:
      block: true
- id: 'properties.paragraph boolean'
  type: Skeleton
  properties:
    paragraph: false
- id: 'properties.paragraph object width all'
  type: Skeleton
  properties:
    paragraph:
      rows: 3
      width: 300
- id: 'properties.paragraph object width array'
  type: Skeleton
  properties:
    paragraph:
      rows: 3
      width:
        - 700
        - 250
        - 500
