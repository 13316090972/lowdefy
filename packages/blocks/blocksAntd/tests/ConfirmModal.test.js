/*
  Copyright 2020-2021 Lowdefy, Inc

  Licensed under the Apache License, Version 2.0 (the "License");
  you may not use this file except in compliance with the License.
  You may obtain a copy of the License at

      http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS IS" BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the License.
*/

import { runBlockSchemaTests, runMockMethodTests } from '@lowdefy/block-dev';
import Enzyme, { mount } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { Modal } from 'antd';

Enzyme.configure({ adapter: new Adapter() });
import ConfirmModal from '../src/blocks/ConfirmModal/ConfirmModal';
import examples from '../demo/examples/ConfirmModal.yaml';
import meta from '../src/blocks/ConfirmModal/ConfirmModal.json';

jest.mock('@lowdefy/block-tools', () => {
  const originalModule = jest.requireActual('@lowdefy/block-tools');
  return {
    ...originalModule,
    blockDefaultProps: {
      ...originalModule.blockDefaultProps,
      methods: {
        ...originalModule.blockDefaultProps.methods,
        makeCssClass: jest.fn((style, op) => JSON.stringify({ style, options: op })),
      },
    },
  };
});

jest.mock('antd/lib/modal', () => {
  return {
    confirm: jest.fn(),
    error: jest.fn(),
    info: jest.fn(),
    success: jest.fn(),
    warning: jest.fn(),
  };
});

const mocks = [
  {
    name: 'confirm',
    fn: Modal.confirm,
  },
  {
    name: 'error',
    fn: Modal.error,
  },
  {
    name: 'info',
    fn: Modal.info,
  },
  {
    name: 'success',
    fn: Modal.success,
  },
  {
    name: 'warning',
    fn: Modal.warning,
  },
];

runMockMethodTests({ examples, Block: ConfirmModal, mocks, meta, enzyme: { mount } });
runBlockSchemaTests({ examples, meta });
