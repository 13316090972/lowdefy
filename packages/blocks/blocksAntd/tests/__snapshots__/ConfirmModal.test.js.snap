// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render default 1`] = `
<div
  id="default_popup"
/>
`;

exports[`Render properties.title 1`] = `
<div
  id="properties.title_popup"
/>
`;

exports[`Test Schema default 1`] = `true`;

exports[`Test Schema default 2`] = `null`;

exports[`Test Schema properties.title 1`] = `true`;

exports[`Test Schema properties.title 2`] = `null`;
